### ISSUE TITLE
[cocoon] Migrate Flutter dashboard to support Wasm compilation

### ISSUE THREAD

**Issue by kevmoo** [flutter/flutter#139170]
- [x] https://github.com/flutter/flutter/issues/139167
- [x] https://github.com/flutter/flutter/issues/139169
- [x] https://github.com/codebird/flutter_app_icons/pull/3 - this package is tiny. Might be better to just inline it?
- [x] https://github.com/firebase/flutterfire/issues/12466

**Comment by ditman** [comment]
I updated `flutter_app_icons` last night too: https://github.com/codebird/flutter_app_icons/pull/3

**Comment by ditman** [comment]
Once `dependabot` updates `google_sign_in_web: 0.12.3` [here](https://github.com/flutter/cocoon/blob/main/dashboard/pubspec.yaml#L21), the dashboard should compile to WASM like so:

<img width="1680" alt="Screenshot 2023-12-08 at 9 42 19 PM" src="https://github.com/flutter/flutter/assets/1255594/9354b481-82d7-4ae9-825b-6807d38b1201">

**Comment by kevmoo** [comment]
YES!!!

**Comment by kevmoo** [comment]
https://pub.dev/packages/flutter_app_icons/changelog – published!

**Comment by ditman** [comment]
> https://pub.dev/packages/flutter_app_icons/changelog – published!

Yes, and cocoon has already picked it up! It's `google_sign_in_web` that took me a little bit longer to merge because of a couple of CI issues, that is the last piece of the puzzle :P

**Comment by ditman** [comment]
`firebase_core_web` still doesn't support `--wasm` compilation: 

```
Target dart2wasm failed: ProcessException: Process exited abnormally with exit code 64:
*NOTE*: Compilation to WasmGC is experimental.
The support may change, or be removed, with no advance notice.


/usr/local/google/home/dit/.pub-cache/hosted/pub.dev/firebase_core_web-2.11.5/lib/src/interop/utils/utils.
dart:14:1: Error: JS interop library 'package:js/js.dart' can't be imported when compiling to Wasm.
Try using 'dart:js_interop' or 'dart:js_interop_unsafe' instead.
import 'package:js/js.dart';
^
/usr/local/google/home/dit/.pub-cache/hosted/pub.dev/firebase_core_web-2.11.5/lib/src/interop/utils/utils.
dart:15:1: Error: JS interop library 'package:js/js_util.dart' can't be imported when compiling to Wasm.
Try using 'dart:js_interop' or 'dart:js_interop_unsafe' instead.
import 'package:js/js_util.dart' as util;
^

... and more ...
```

**Comment by kevmoo** [comment]
Should be easy to update the pin, no?

**Comment by ditman** [comment]
> Should be easy to update the pin, no?

@kevmoo dependabot should be doing it (?)

* https://github.com/flutter/cocoon/blob/9203992c85dc74a8aa79729dbcd4a98819af945e/dashboard/pubspec.yaml#L17-L18

(we need `firebase_core: ^2.29.0`, I think)

**Comment by ditman** [comment]
I haven't attempted this in a while!

**Comment by rayliverified** [comment]
google_sign_in might be broken on wasm.

I'll make a test repo tonight.

**Comment by kevmoo** [comment]
@rayliverified – yeah we're tracking https://github.com/firebase/flutterfire which might be related

**Comment by rayliverified** [comment]
Ah, I didn't find it earlier.

This one here: https://github.com/firebase/flutterfire/issues/12819

Do you know of the Firebase Storage issue with not being able to upload Uint8List data? I can file that one if it doesn't exist.

**Comment by kevmoo** [comment]
> Ah, I didn't find it earlier.
> 
> This one here: [firebase/flutterfire#12819](https://github.com/firebase/flutterfire/issues/12819)

sounds related

> Do you know of the Firebase Storage issue with not being able to upload Uint8List data? I can file that one if it doesn't exist.

I don't! Please CC me!

**Comment by ditman** [comment]
> google_sign_in might be broken on wasm.
> 
> I'll make a test repo tonight.

@rayliverified did you end up creating an issue to `google_sign_in_web` and wasm?

**Comment by rayliverified** [comment]
Firebase storage is fixed, thank you all!

Google sign in Web is still broken. I think there's an issue tracking it
made by someone else maybe.

**Comment by ditman** [comment]
Ah, I thought it was a firebase problem, but it seems it's a Google Sign In issue, hmm... I'll dive deeper, thanks for bringing this to my attention @rayliverified, I totally had missed that github issue!

**Comment by kevmoo** [comment]
This is already done!
